pkg/s3wrapper/client.go

```diff
@@ -341,3 +346,70 @@ func (c *S3Client) handleObject(ctx context.Context, log logrus.FieldLogger, obj
 		}
 	}
 }
+
+func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]string, error) {
+	log := logutil.FromContext(ctx, c.log)
+	var objects []string
+	log.Info("Listing objects by with prefix %s", prefix)
+	resp, err := c.client.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String(c.cfg.S3Bucket),
+		Prefix: aws.String(prefix),
+	})
+	if err != nil {
+		err = errors.Wrapf(err, "Error listing objects for prefix %s", prefix)
+		log.Error(err)
+		return nil, err
+	}
+	for _, key := range resp.Contents {
+		objects = append(objects, *key.Key)
+	}
+	return objects, nil
+}
+
+func (c *S3Client) DownloadListOfFiles(ctx context.Context, files []string) (io.ReadCloser, int64, error) {
+	log := logutil.FromContext(ctx, c.log)
+	// Create a new stream.
+	w, err := stream.New("downloads")
+	if err != nil {
+		err = errors.Wrapf(err, "Failed tp create stream for files %s", files)
+		log.Error(err)
+		return nil, 0, err
+	}
+	defer w.Close()
+
+	// Create a new tar archive.
+	tarWriter := tar.NewWriter(w)
+	var rdr io.ReadCloser
+	var objectSize int64
+
+	// Create tar headers from s3 files
+	for _, file := range files {
+
+		// Read file from S3, log any errors
+		rdr, objectSize, err = c.Download(ctx, file)
+		if err != nil {
+			log.WithError(err).Warnf("Failed to open reader for %s, skipping it", file)
```

Comment:
The file should exists because you just did a list files a moment ago, so what can be the error here?
i'm not sure if you want to skip files on unknown errors. 
It can be a confusing ux that user try to download logs and get an empty list of files and API return OK.

Reply:
i just wanted to return all i can

Reply:
When we;ll have retention on these files, they might be deleted in the middle, so I think we do have to handle the case

Reply:
i think that you should return an error. User will not know that he doesn't have all the logs and i can be confusing.

Reply:
u think we need to fail here?